File: typescript-client-sdk/src/sdk/models/operations/puteventseventid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform, Type } from "class-transformer";


export class PutEventsEventIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=event_id" })
  eventId: string;
}

export class PutEventsEventIdRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "customer_id" })
  customerId?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "event_name" })
  eventName: string;

  @SpeakeasyMetadata()
  @Expose({ name: "external_customer_id" })
  externalCustomerId?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "properties" })
  properties: Record<string, any>;

  @SpeakeasyMetadata()
  @Expose({ name: "timestamp" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  timestamp: Date;
}

export class PutEventsEventIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PutEventsEventIdPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutEventsEventIdRequestBody;
}

export class PutEventsEventId400ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "title" })
  title?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "validation_errors" })
  validationErrors?: string[];
}

export class PutEventsEventId200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "amended" })
  amended?: string;
}

export class PutEventsEventIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  putEventsEventId200ApplicationJSONObject?: PutEventsEventId200ApplicationJSON;

  @SpeakeasyMetadata()
  putEventsEventId400ApplicationJSONObject?: PutEventsEventId400ApplicationJSON;
}

File: go-client-sdk/pkg/models/operations/puteventseventid.go
package operations

import (
	"time"
)

type PutEventsEventIDPathParams struct {
	EventID string `pathParam:"style=simple,explode=false,name=event_id"`
}

type PutEventsEventIDRequestBody struct {
	CustomerID         *string                `json:"customer_id,omitempty"`
	EventName          string                 `json:"event_name"`
	ExternalCustomerID *string                `json:"external_customer_id,omitempty"`
	Properties         map[string]interface{} `json:"properties"`
	Timestamp          time.Time              `json:"timestamp"`
}

type PutEventsEventIDRequest struct {
	PathParams PutEventsEventIDPathParams
	Request    *PutEventsEventIDRequestBody `request:"mediaType=application/json"`
}

type PutEventsEventID400ApplicationJSON struct {
	Status           *int64   `json:"status,omitempty"`
	Title            *string  `json:"title,omitempty"`
	Type             *string  `json:"type,omitempty"`
	ValidationErrors []string `json:"validation_errors,omitempty"`
}

type PutEventsEventID200ApplicationJSON struct {
	Amended *string `json:"amended,omitempty"`
}

type PutEventsEventIDResponse struct {
	ContentType                              string
	StatusCode                               int
	PutEventsEventID200ApplicationJSONObject *PutEventsEventID200ApplicationJSON
	PutEventsEventID400ApplicationJSONObject *PutEventsEventID400ApplicationJSON
}
